### DISCUSSION TITLE
How to refactor a diagram?

### DISCUSSION THREAD

**Discussion by jbreitbart** [terrastruct/d2#556] [General]
I am currently playing around with d2 via the obsidian plugin and I have a somewhat nested diagram that I am changing a lot. Is there any IDE like support of renaming symbols or support for moving elements inside another element to make layering easier?

**Comment by alixander** [comment]
not yet. we plan to kick off an LSP project, but may take a bit before getting to it (https://github.com/terrastruct/d2/issues/104)

**Comment by jbreitbart** [comment]
Thanks, definitionally useful 😄
